test/sys/test_sockopt.rs

```diff
@@ -1097,6 +1097,36 @@ mod sockopt_impl {
         assert_eq!(get_linger.l_linger, set_linger.l_linger);
     }
 
+    #[cfg(apple_targets)]
+    sockopt_impl!(
+        LingerSec,
+        Both,
+        libc::SOL_SOCKET,
+        libc::SO_LINGER_SEC,
+        libc::linger
+    );
```

Comment:
Looks like you forgot to remove this:)

Reply:
No I did the same than with the above test.

Reply:
Emmm, why do we need to define it here?

Reply:
I can try to "fix" both tests if you like ?

Reply:
Hmm, this module `mod sockopt_impl {` is for testing the `sockopt_impl!` macro, we made it and related stuff public in #2556 so that users can define socket options without contributing to Nix. We should not put the `fn test_linger_sec()` test here

Reply:
Please put the test [before that module](https://github.com/nix-rust/nix/blob/e789a7cd2d6b21d8962de482b9a5da412b66017a/test/sys/test_sockopt.rs#L1065), and if so, I guess we do not need to re-define the `LingerSec` socket option